spec/services/invoices/one_off_service_spec.rb

```diff
@@ -0,0 +1,209 @@
+      expect do
+        invoice_service.create
+      end.not_to have_enqueued_job(ActionMailer::MailDeliveryJob)
+    end
+
+    context 'with lago_premium' do
+      around { |test| lago_premium!(&test) }
+
+      it 'enqueues an ActionMailer::MailDeliveryJob' do
+        expect do
+          invoice_service.create
+        end.to have_enqueued_job(ActionMailer::MailDeliveryJob)
+      end
+
+      context 'when organization does not have right email settings' do
+        before { customer.organization.update!(email_settings: []) }
+
+        it 'does not enqueue an ActionMailer::MailDeliveryJob' do
+          expect do
+            invoice_service.create
+          end.not_to have_enqueued_job(ActionMailer::MailDeliveryJob)
+        end
+      end
+    end
+
+    context 'when organization does not have a webhook url' do
+      before { customer.organization.update!(webhook_url: nil) }
+
+      it 'does not enqueues a SendWebhookJob' do
+        expect do
+          invoice_service.create
+        end.not_to have_enqueued_job(SendWebhookJob)
+      end
+    end
+
+    context 'with customer timezone' do
+      before { customer.update!(timezone: 'America/Los_Angeles') }
+
+      let(:timestamp) { DateTime.parse('2022-11-25 01:00:00') }
+
+      it 'assigns the issuing date in the customer timezone' do
+        result = invoice_service.create
+
+        expect(result.invoice.issuing_date.to_s).to eq('2022-11-24')
+      end
+    end
+
+    context 'when currency does not match' do
+      let(:currency) { 'NOK' }
+
+      it 'fails' do
+        result = invoice_service.create
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::ValidationFailure)
+          expect(result.error.messages.keys).to include(:currency)
+          expect(result.error.messages[:currency]).to include('currencies_does_not_match')
+        end
+      end
+    end
+
+    context 'when customer is not found' do
+      let(:customer) { nil }
+
+      it 'returns a not found error' do
+        result = invoice_service.create
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::NotFoundFailure)
+          expect(result.error.message).to eq('customer_not_found')
+        end
+      end
+    end
+
+    context 'when fees are blank' do
+      let(:fees) { [] }
+
+      it 'returns a not found error' do
+        result = invoice_service.create
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::NotFoundFailure)
+          expect(result.error.message).to eq('fees_not_found')
+        end
+      end
+    end
+
+    context 'when add_on_code is invalid' do
+      let(:fees) do
+        [
+          {
+            add_on_code: add_on_first.code,
+            unit_amount_cents: 1200,
+            units: 2,
+            description: 'desc-123'
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.